napari/util/io.py

```diff
@@ -62,9 +63,19 @@ def magic_read(filenames, *, use_dask=None, stack=True):
     for filename in filenames_expanded:
         ext = os.path.splitext(filename)[-1]
         if ext == '.zarr':
-            image = da.from_zarr(filename)
-            if shape is None:
-                shape = image.shape
+            zr = zarr.open(filename, mode='r')
+            if isinstance(zr, zarr.core.Array):
+                # load zarr array
+                image = da.from_zarr(filename)
+                if shape is None:
+                    shape = image.shape
+            else:
+                # else load zarr all arrays inside file
+                image = [
+                    da.from_zarr(filename, component=c) for c, a in zr.arrays()
+                ]
```

Comment:
What's the story with this change? This is a zarr dataset?

Reply:
If so... This clause is getting kinda big and clunky, maybe let's add sub-function here, `read_zarr_dataset`?

Reply:
this now supports out image pyramid format - it's getting at the difference between a zarr array and a zarr group - the first clause of the if statement applies when the top level object in the zarr file is just a zarr array, the second clause comes into play when it is a zarr group. dask_array.from_zarr can only handle dask arrays

Example of zarr array at top:
![image](https://user-images.githubusercontent.com/6531703/67153359-689e9c80-f2ad-11e9-9834-c11239f97bab.png)

Example of zarr group at top:
![image](https://user-images.githubusercontent.com/6531703/67153369-8d930f80-f2ad-11e9-9169-71e0705814e5.png)

Reply:
i've pulled this out into it's own function